CIP-XXX/CIP-XXX.md

```diff
@@ -0,0 +1,1408 @@
+
+-- Even when mixed with valid ones
+builtinSetBits [0xFF] [(0, False), ((-1), True)] => error
+
+-- In any position
+builtinSetBits [0xFF] [((-1), True), (0, False)] => error
+
+-- Out-of-bounds indexes fail
+builtinSetBits [0xFF] [(8, False)] => error
+
+-- Even when mixed with valid ones
+builtinSetBits [0xFF] [(1, False), (8, False)] => error
+
+-- In any position
+builtinSetBits [0xFF] [(8, False), (1, False)] => error
+
+-- Bits are written 'from the end'
+builtinSetBits [0xFF] [(0, False)] => [0xFE]
+
+builtinSetBits [0xFF] [(1, False)] => [0xFD]
+
+builtinSetBits [0xFF] [(2, False)] => [0xFB]
+
+builtinSetBits [0xFF] [(3, False)] => [0xF7]
+
+builtinSetBits [0xFF] [(4, False)] => [0xEF]
+
+builtinSetBits [0xFF] [(5, False)] => [0xDF]
+
+builtinSetBits [0xFF] [(6, False)] => [0xBF]
+
+builtinSetBits [0xFF] [(7, False)] => [0x7F]
+
+-- True value sets the bit
+builtinSetBits [0x00] [(5, True)] => [0x20]
+
+-- False value clears the bit
+builtinSetBits [0xFF] [(5, False)] => [0xDF]
+
+-- Larger indexes write backwards into the bytes from the end
+builtinSetBits [0xF4, 0xFF] [(10, True)] => [0xF0, 0xFF]
+
+-- Multiple items in a change list apply cumulatively
+builtinSetBits [0xF4, 0xFF] [(10, True), (1, False)] => [0xF0, 0xFD]
+
+builtinSetBits (builtinSetBits [0xF4, 0xFF] [(10, True)]) [(1, False)] => [0xF0, 0xFD]
+
+-- Order within a changelist is unimportant among unique indexes
+builtinSetBits [0xF4, 0xFF] [(1, False), (10, True)] => [0xF0, 0xFD]
+
+-- But _is_ important for identical indexes
+builtinSetBits [0x00, 0xFF] [(10, True), (10, False)] => [0x00, 0xFF]
+
+builtinSetBits [0x00, 0xFF] [(10, False), (10, True)] => [0x04, 0xFF]
+
+-- Setting an already set bit does nothing
+builtinSetBits [0xFF] [(0, True)] => [0xFF]
+
+-- Clearing an already clear bit does nothing
+builtinSetBits [0x00] [(0, False)] => [0x00]
+```
+
+#### `builtinReplicate`
+
+`builtinReplicate` takes two arguments; we name and describe them below.
+
+1. The desired result length, of type `BuiltinInteger`. This is the _length
+   argument_.
+2. The byte to place at each position in the result, represented as a
+   `BuiltinInteger` (corresponding to the unsigned integer this byte encodes).
+   This is the _byte argument_.
+
+Let $n$ be the length argument, and $w$ the byte argument. If $n < 0$, then
+`builtinReplicate` fails. In this case, the resulting error message must specify
+_at least_ the following information:
+
+* That `builtinReplicate` failed due to a negative length argument; and
+* What `BuiltinInteger` was passed as the length argument.
+
+If $n \geq 0$, and $w < 0$ or $w > 255$, then `builtinReplicate` fails. In this
+case, the resulting error message must specify _at least_ the following
+information:
+
+* That `builtinReplicate` failed due to the byte argument not being a valid
+  byte; and
+* What `BuiltinInteger` was passed as the byte argument.
+
+Otherwise, let $b$ be the result of `builtinReplicate`, and let $b\\{i\\}$ be the
```

Comment:
```suggestion
Otherwise, let $b$ be the result of `builtinReplicate`, and let $b\\[i\\]$ be the
```
worth checking other uses for consistency. Shame we don't have macros...

Reply:
I deliberately used `{}` here, as I'm indexing bytes, not bits, which is what `[]` is used for.